### DISCUSSION TITLE
Unreliable wakeup when using Waveshare SIM7600X board

### DISCUSSION THREAD

**Discussion by leighghunt** [PiSupply/PiJuice#990] [General]
Hi, I'm using a number of PiJuices in a remote camera setup (timelapse, and monitoring pollution on building sites - https://github.com/venari/timelapse).

I've had issues with power draw from a HUAWEI 4G Wingle E8372, so I'm looking to move to a HAT to provide network access.

Whilst testing a board to check it all works with the PiJuice/Waveshare SIM7600X/Pi Zero setup, I've noticed that when the SIM7600 hat is attached, the wake up becomes unreliable.

More info:

I've got 12 cameras out in the field and they're reliably waking up. I have them shutdown overnight and wake up every hour. During the day they run for 12 hours continuously. In addition, I have a process that detects failure to upload or capture imagery and bounce the Pi if this occurs. I do this by setting a wake up 5 minutes in the future and do a full power down to try and give everything a chance to reset.

With my test unit on my bench, over the course of a few days testing I'm finding the following:

| Pi Juice USB Power | 7600X Hat       | Comments       |
|--------------------|--------------|----------------|
| Attached           | Attached     | Wakes reliably |
| Not attached       | Not attached | Wakes reliably |
| Attached           | Not attached | Wakes reliably |
| Not attached       | Attached     | Wakes intermittently   |

Instead of regularly restarting every 15 minutes when either powered via PiJuice USB, or with the 7600X not attached, it's like it is missing the wake up alarm - examples I've seen are 7 hours late, 1 or 2 hours late, and 5 hours late, all on the right minute.

From my log files and reviewing system log files, it appears that the Pi is not powering up at all during these periods. I've got the RTC overlays set and clock is working reliably.

So my question is whether use of the PiJuice and Waveshare SIM7600X board should work in this scenario, or if there's is anything I need to change to make the wake up alarm more reliable. I've seen mention in other tickets of capacitance of other boards affecting wake up alarms - could this be part of the issue?

I've just seen reference to the `pijuice_log` script, so will gather some of those results to see how they differ between scenarios, I'll attach these in the next 24 hours.

**Comment by leighghunt** [comment]
Some more info... I enabled the pijuice_log script:

```
python3 /usr/bin/pijuice_log.py --enable WAKEUP_EVT
```

```
0 WAKEUP_EVT   2023-07-17 00:46:00.898437, Battery: 96.0%, 4.174V, 36C, CHARGING_FROM_IN
        GPIO_5V: REGULATOR: OFF, 0.505V, 0.000A, NOT_PRESENT
        TRIGGERS:  RTC
        WAKEUP_ON_CHARGE: DISABLED

1 WAKEUP_EVT   2023-07-17 01:19:00.652343, Battery: 98.4%, 4.171V, 35C, CHARGING_FROM_IN
        GPIO_5V: REGULATOR: OFF, 0.486V, 0.000A, NOT_PRESENT
        TRIGGERS:  RTC
        WAKEUP_ON_CHARGE: DISABLED

2 WAKEUP_EVT   2023-07-17 01:52:00.992187, Battery: 78.4%, 3.951V, 33C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.482V, 0.000A, NOT_PRESENT
        TRIGGERS:  RTC
        WAKEUP_ON_CHARGE: DISABLED

3 WAKEUP_EVT   2023-07-17 02:52:00.992187, Battery: 73.6%, 3.951V, 30C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.388V, 0.000A, NOT_PRESENT
        TRIGGERS:  RTC
        WAKEUP_ON_CHARGE: DISABLED

4 WAKEUP_EVT   2023-07-17 03:52:00.996093, Battery: 73.6%, 3.951V, 31C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.378V, 0.000A, NOT_PRESENT
        TRIGGERS:  RTC
        WAKEUP_ON_CHARGE: DISABLED

5 WAKEUP_EVT   2023-07-17 03:56:42.972656, Battery: 73.6%, 3.948V, 30C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.378V, 0.000A, NOT_PRESENT
        TRIGGERS: POWER_BUTTON
        WAKEUP_ON_CHARGE: DISABLED

6 WAKEUP_EVT   2023-07-17 03:56:59.375000, Battery: 73.6%, 3.948V, 31C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.399V, 0.000A, NOT_PRESENT
        TRIGGERS: POWER_BUTTON
        WAKEUP_ON_CHARGE: DISABLED

7 WAKEUP_EVT   2023-07-17 03:56:59.054687, Battery: 73.6%, 3.948V, 31C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.955V, 0.000A, NOT_PRESENT
        TRIGGERS: POWER_BUTTON
        WAKEUP_ON_CHARGE: DISABLED

8 WAKEUP_EVT   2023-07-17 03:57:36.113281, Battery: 73.6%, 3.855V, 30C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.386V, 0.000A, NOT_PRESENT
        TRIGGERS: POWER_BUTTON
        WAKEUP_ON_CHARGE: DISABLED

9 WAKEUP_EVT   2023-07-17 03:57:48.613281, Battery: 73.6%, 3.944V, 31C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.421V, 0.000A, NOT_PRESENT
        TRIGGERS: POWER_BUTTON
        WAKEUP_ON_CHARGE: DISABLED

10 WAKEUP_EVT   2023-07-17 03:57:48.273437, Battery: 73.6%, 3.948V, 31C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.934V, 0.000A, NOT_PRESENT
        TRIGGERS: POWER_BUTTON
        WAKEUP_ON_CHARGE: DISABLED

11 WAKEUP_EVT   2023-07-17 03:57:59.453125, Battery: 73.6%, 3.923V, 31C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.425V, 0.000A, NOT_PRESENT
        TRIGGERS: POWER_BUTTON
        WAKEUP_ON_CHARGE: DISABLED

12 WAKEUP_EVT   2023-07-17 03:58:00.992187, Battery: 73.6%, 3.898V, 31C, NORMAL
        GPIO_5V: REGULATOR: OFF, 0.821V, 0.000A, NOT_PRESENT
        TRIGGERS: POWER_BUTTON
        WAKEUP_ON_CHARGE: DISABLED
```

I've overlaid these events over my telemetry (in NZ time, so hours out by 12) - you can see that WAKEUP_EVT's 0 and 1 fire correctly (restarting the Pi after 30 minutes of no image capture as expected whilst testing my script running with no camera connected). 
At 1:30 I unplug the PiJuice USB power connection (you can see the battery start to discharge), and you can then see that events 2, 3, and 4 fire, but don't cause the Pi to wake. If I left this for a few hours, it would probably wake up.
I then pressed SW1 a few times, events 5 - 12 (1 press, then double press, then 20 second hold down, then a press, and then a double press) at which point it powered up.

<img width="1325" alt="Screen Shot 2023-07-17 at 4 03 46 PM" src="https://github.com/PiSupply/PiJuice/assets/349153/6abc861b-c6b1-49d7-800e-5fd3d87099fc">

**Comment by leighghunt** [comment]
Not meaning to talk too soon, but I've switched the PiJuice for a new straight out-of-the-box PiJuice meaning firmware changes from 1.6 -> 1.5, and the battery changes from my custom profile 10,000mAh one to `PJZERO_1000`.

It's restarted reliably now three times in a row, which it's not done previously. I'll leave it running overnight and see if that's addressed the issue and report back.

<img width="1323" alt="Screen Shot 2023-07-17 at 6 20 31 PM" src="https://github.com/PiSupply/PiJuice/assets/349153/35751bc8-8835-4fb9-98fc-d21c6e0947e8">

**Comment by leighghunt** [comment]
Follow up - the reliable wake ups continued for ten in a row until I tweaked the battery profile from whatever the default was (`PJZERO_1000`?) to `PJLIP0_12000`. It then carried on through the night restarting every 10 minutes and shutting down - as my code does this in a low battery situation.

I also tweaked the temperature sensing.

The next day, I charged the PiJuice, and when I disconnected it, it stopped being reliable again.

I'm going to try and reset the PiJuice, or use another new one, and see what triggered the waking up to become unreliable - but suspect it's one of:

- Battery profile changing
- SOC setting
- Temperature setting
Firmware has not been updated and is v1.5.
<img width="1313" alt="Screen Shot 2023-07-18 at 3 49 08 PM" src="https://github.com/PiSupply/PiJuice/assets/349153/79ad9931-7177-49a7-af8d-fe858fb63ccf">

**Comment by tvoverbeek** [comment]
You write you changed the profile to the 12000 battery.
Note: there is a batch of 12000 mAh batteries which have a different NTC (for measuring battery temp).
If you have one of those it might explain the problems you have. Same for 1.5 and 1.6 firmwares.
Suggest you use the custom profile (starting from the 12000 profile) and use the NTC values specified here:
https://github.com/PiSupply/PiJuice/blob/master/Software/Configs/Batteries/LIPO_12000_PIS-1129.JSON
Note: NTC resistance value here is 42000 (420k) instead of 1000 (10k) in the firmware 12000 profile and
          the ntcB constant is 5200 instead of 3450 in the firmware 12000 profile.

**Comment by leighghunt** [comment]
> You write you changed the profile to the 12000 battery. Note: there is a batch of 12000 mAh batteries which have a different NTC (for measuring battery temp). If you have one of those it might explain the problems you have. Same for 1.5 and 1.6 firmwares. Suggest you use the custom profile (starting from the 12000 profile) and use the NTC values specified here: https://github.com/PiSupply/PiJuice/blob/master/Software/Configs/Batteries/LIPO_12000_PIS-1129.JSON Note: NTC resistance value here is 42000 (420k) instead of 1000 (10k) in the firmware 12000 profile and the ntcB constant is 5200 instead of 3450 in the firmware 12000 profile.

Do you think it could be the battery itself? The battery I'm using (and I appreciate this isn't supported, but seems to work fine without the SIM7600X installed) is a set of 4 x 18650 2600mAh batteries in parallel, with no temp monitoring. I can swap this for a supported battery instead - I just so happen to have ordered yesterday 3 of the 12,000mAh PiSupply batteries (first time I've managed to get them shipped to NZ), so can do the test with that and the custom profiles you mention when it arrives.

More info - I did the SW2/SW1 reset to the PiJuice but the wakeup is still unreliable - failed to wake up overnight at all. Woke up correctly once power supplied to PiJuice micro USB socket. 

I've got two more new PiJuices so I'll do some more testing whilst I wait for the batteries to arrive.

**Comment by leighghunt** [comment]
Yay! 3 new shiny 12,000 mAh batteries arrived from PiSupply today, so I'll do some more testing with v1.5 and v1.6 firmwares, and will report back later this week.

![image](https://github.com/PiSupply/PiJuice/assets/349153/6e4523c0-c996-43a4-8959-1d850b646069)

**Comment by leighghunt** [comment]
Update - I've been testing this with the PiJuice 12,000 mAh battery for the last week with both v1.5 and v1.6 battery firmware, and it hasn't skipped a beat. Wake-ups have been perfectly reliable.

And as an added bonus, battery performance has exceeded both the original 15Ah LiFePO4 batteries I've got in the field, and my home brew 4x18650 in parallel concoction.

So I guess I don't understand exactly the mechanism, but I suspect the root cause of the issue is more likely to be my original unsupported battery and will close this discussion - but leave it here in case anyone else has similar symptoms.

**Comment by leighghunt** [comment]
Last update - everything has been working flawlessly the last couple of weeks.

**Comment by leighghunt** [comment]
For anyone else who finds this discussion and is experiencing similar behaviour, I've had this occur with another PiJuice, this time with a LiFePO4 battery, and not using the Waveshare modem.

The Pi was seemingly not waking when not connected to an external power source - but would work fine if already booted.

I think it was actually down to a poor power connection over a piece of wire that was too long - whilst troubleshooting I noticed that there was a large voltage drop occurring on the negative wire back to the battery - I think the Pi was actually booting up but not managing to complete the boot, as the process of booting was drawing a little too much current. I've shortened the cable and reconnected and it's since working better. Something else to eliminate if you're having strange wake up issues.

**Comment by leighghunt** [comment]
A further update - I had 12 cameras mostly working well for the next 6-7 months across a summer and winter. However there were still occasional wake up issues and a number of random restarts (worked around with the watchdog functionality).

I tweaked the design and dropped the Waveshare modem and switched back to the HUAWEI thumb modem stick powered from VSYS via the System Power Switch, and this combined with the LiPo rather than LiFePO4 battery seems to have addressed the wakeup and restart issues. Power use also seems better now that I no longer needed to use a buck boost convertor to up the VSYS from 3.5-3.8V to 5 for USB (not needed with LiPo)

So to anyone else using a PiJuice in the field with cellular modem, I would suggest avoiding LiFePO4 battery chemistry and also maybe not using the Waveshare board I was using.
